frontend/src/org/jetbrains/common/coroutines.kt

```diff
@@ -0,0 +1,21 @@
+package kotlinx.coroutines.experimental
+
+import kotlin.coroutines.experimental.*
+import kotlin.js.Promise
+
+suspend fun <T> Promise<T>.await() = suspendCoroutine<T> { cont ->
+    then({ value -> cont.resume(value) },
+        { exception -> cont.resumeWithException(exception) })
+}
+
+fun <T> async(block: suspend () -> T): Promise<T> = Promise<T> { resolve, reject ->
```

Comment:
In short - yes. 

The longer answer is that `async` is supposed to be used for writing async-style (promise-returning) functions, which are not actually needed anywhere in this problem. So, I've started the port to coroutines with `launch` only, but then the resulting `try { ... } catch (e: Throwable) { ... }` code looked uglier than the original promise-based `... .catch { e -> ... }` code, so I've started using `async` as a visually more pleasing exception handling aid. 

In fact, this problem should have never been present, because, as you can see, exception handling in all components here is similar. The next obvious refactoring is to extract exception handling into some common code.